apps/examples/fs_test/smartfs_test.c

```diff
@@ -0,0 +1,329 @@
+{
+	int ret;
+	long time;
+	long timediff;
+
+	time = get_time();
+	ret = unlink(filepath);
+	if (ret < 0) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		return ERROR;
+	}
+	timediff = get_time() - time;
+	g_total_time += timediff;
+	print_log("remove:%dms, ", timediff);
+
+	return OK;
+}
+
+static int write_read_test(char *filepath, void *buf, int nbytes, int iteration)
+{
+	int i;
+	int ret;
+
+	for (i = 0; i < iteration; i++) {
+		ret = fs_write(filepath, buf, nbytes);
+		if (ret < 0) {
+			return ERROR;
+		}
+
+		ret = fs_read(filepath, buf, nbytes);
+		if (ret < 0) {
+			return ERROR;
+		}
+	}
+
+	return OK;
+}
+
+static int write_test(char *filepath, void *buf, int nbytes, int iteration)
+{
+	int i;
+	int ret; 
+
+	for (i = 0; i < iteration; i++) {
+		ret = fs_write(filepath, buf, nbytes);
+		if (ret < 0) {
+			return ERROR;
+		}
+	}
+
+	return OK;
+}
+
+static int read_test(char *filepath, void *buf, int nbytes, int iteration)
+{
+	int i;
+	int ret;
+
+	ret = fs_write(filepath, buf, nbytes);
+	if (ret < 0) {
+		return ERROR;
+	}
+
+	for (i = 0; i < iteration; i++) {
+		ret = fs_read(filepath, buf, nbytes);
+		if (ret < 0) {
+			return ERROR;
+		}
+	}
+
+	return OK;
+}
+
+static int write_remove_test(char *filepath, void *buf, int nbytes, int iteration)
+{
+	int i;
+	int ret;
+
+	for (i = 0; i < iteration; i++) {
+		ret = fs_write(filepath, buf, nbytes);
+		if (ret < 0) {
+			return ERROR;
+		}
+
+		ret = fs_remove(filepath, buf, nbytes);
+		if (ret < 0) {
+			return ERROR;
+		}
+	}
+
+	return OK;
+}
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int smartfs_test_main(int argc, char *argv[])
+#endif
+{
+	int ret;
+	int size;
+	int operation;
+	int iteration;
+	char *buf;
+	char *filepath;
+
+	if (argc == 6) {
+		printf("FS R/W Performance Test!\n");
+
+		operation = atoi(argv[1]);
+		filepath = argv[2];
+		size = atoi(argv[3]);
+		iteration = atoi(argv[4]);
+		g_printopt = argv[5][0];
+
+		if (operation < TEST_OPT_WRITEREAD || operation >= TEST_OPT_MAX ||
+			size < 0 || iteration < 1 || iteration > FS_RW_ITERATION_MAX) {
+			printf("%s", USAGE);
+			return ERROR;
+		}
+
+		if (strlen(argv[5]) != 1 || (g_printopt != 'y' && g_printopt != 'n')) {
+			printf("%s", USAGE);
+			return ERROR;
+		}
+
+		buf = (char *)malloc(size);
+		if (!buf) {
+			printf("%s: Malloc failed, %d\n", __FUNCTION__, size);
+			return ERROR;
+		}
+
+		memset(buf, 0, size);
+
+		switch (operation) {
+		case TEST_OPT_WRITEREAD:
+			ret = write_read_test(filepath, buf, size, iteration);
+			break;
+		case TEST_OPT_WRITE:
+			ret = read_test(filepath, buf, size, iteration);
+			break;
+		case TEST_OPT_READ:
+			ret = write_test(filepath, buf, size, iteration);
+			break;
+		case TEST_OPT_WRITEREMOVE:
+			ret = write_remove_test(filepath, buf, size, iteration);
+			break;
+		}
+		free(buf);
+
+		if (ret == OK) {
+			/* Print measured run-time on success */
+			printf("run_time: %ld ms\n", g_total_time);
+		}
+		return OK;
```

Comment:
Regardless of ret, it returns OK always. I asked you to add failure message. This is not what I want and this is wrong.